#goal
float64 amplitude
float64 frequency
float64 cycles
---
#result
bool completion
---
#feedback
bool feedback